python_modules/dagster/dagster/core/object_store.py

```diff
@@ -93,8 +93,10 @@ def set_value(self, obj, context, runtime_type, paths):
         return self.set_object(obj, context, runtime_type, paths)
 
     def get_value(self, context, runtime_type, paths):
-        if runtime_type in self.TYPE_REGISTRY:
-            return self.TYPE_REGISTRY[runtime_type].get_object(self, context, runtime_type, paths)
+        if runtime_type.name is not None and runtime_type.name in self.TYPE_REGISTRY:
```

Comment:
we might consider hard throwing when name is None since that is explicitly not working right now and then linking to issue in the exception error message

Reply:
we can't catch that at this level, we would need to look recursively through runtime_type.inner_types to see if we found something in the registry

Reply:
added #1057 to track